rilldata/rill
# Reduce excessive calls to `/toplist`

## Checklist

## Summary
#### Issue addressed: 
[Notion page](https://www.notion.so/rilldata/Only-run-queries-once-dashboard-state-is-initialized-451de063dba34117a4cccfdc189e17ae?d=ac03ba48ef4d45ac865bc46068a7164e)

#### Details:
This PR follows #2563. It adds the `enabled` flag to our TanStack Query calls to `/toplist`. This eliminates the extra requests that did not include a time range and would result in "all time" queries.

This PR does not totally eliminate extra calls to `/toplist`. Here are a couple observations that can be fixed later 
- The comparisons code iteratively builds up a `comparisonFilterSet` object, and queries are still firing as that object is built iteratively.
- At times, custom time ranges produce extra queries, similar to the behavior we observed in PR #2563.

## Steps to Verify
1. Open a dashboard with 1 dimension
2. Open your browser devtools
3. Go to the Network tab
4. Filter for `/toplist`
5. Refresh the dashboard page and observe 1 request to `/toplist` and that request has a specified time range. Previously, there were extra requests to `/toplist`, some of which did not have a specified time range.
